dash-renderer/src/observers/loadingMap.ts

```diff
@@ -0,0 +1,77 @@
+import {
+    equals,
+    flatten,
+    forEach,
+    isEmpty,
+    map,
+    reduce
+} from 'ramda';
+
+import { setLoadingMap } from '../actions/loadingMap';
+import { IStoreObserverDefinition } from '../StoreObserver';
+import { IStoreState } from '../store';
+
+const observer: IStoreObserverDefinition<IStoreState> = {
+    observer: ({
+        dispatch,
+        getState
+    }) => {
+        const {
```

Comment:
This observer sets the store value `loadingMap` and from the TreeContainer's perspective, replaces `pendingCallbacks` as the mean to determine the loading state of a component and its sub-layout.

This is done by constructing a "tree" where each node corresponds to a path-fragment. Each node contains child nodes for continuing path-fragments and a list of all the (id, prop) tuples of callback outputs that could affect the node itself or one of its children.

For example,
Callback #1 may affect (id=a,prop=value) at ["props", "children", 0]
Callback #2 may affect (id=b,prop=value) at ["props", "children", 1] and (id=b,prop=text) at ["props", "children", 1]

The resulting tree will be
```
{
  "__dashprivate__idprop__": [(id=a,prop=value),(id=b,prop=value),(id=b,prop=text)]
  "props": {
      "__dashprivate__idprop__": [(id=a,prop=value),(id=b,prop=value),(id=b,prop=text)]
    "children": {
        "__dashprivate__idprop__": [(id=a,prop=value),(id=b,prop=value),(id=b,prop=text)]
      0: {
          "__dashprivate__idprop__": [(id=a,prop=value)]
      },
      1: {
          "__dashprivate__idprop__": [(id=b,prop=value),(id=b,prop=text)]
      }
    }
  }
}
```